### PR TITLE
feat: Add instance flavor and size arguments for det deploy aws [INFENG-227]

### PR DESCRIPTION
**Pull request by julian-determined-ai**
## Description

Adds the ability to specify a database instance class and initial storage size for a `simple-rds` deployment when using `det deploy aws`.

 

## Test Plan

1. Run `det deploy aws up --deployment-type simple-rds --db-size <size> --db-class <class> ...` and ensure the chosen instance size and classes were used in the resulting cluster. 

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
INFENG-227

### CODE REVIEW

**Review by loksonarius — APPROVED** [review]

File: harness/determined/deploy/aws/constants.py [hunk 1]
```diff
@@ -15,6 +15,8 @@ class defaults:
     REGION = "us-west-2"
     STACK_TAG_KEY = "managed-by"
     STACK_TAG_VALUE = "determined"
+    DB_SIZE = 20
+    DB_CLASS = "db.m7g.large"
```
Comment (tayritenour) [inline harness/determined/deploy/aws/constants.py:19]: I would prefer if we were to rename this to `DB_INSTANCE_CLASS` so it's more clear what this is without having to check the defaults

File: harness/determined/deploy/aws/templates/simple-rds.yaml [hunk 2]
```diff
@@ -99,6 +99,14 @@ Parameters:
...
     Description: Password for database (eg. "postgres")
     NoEcho: true
+  
+  DBSize:
+    Type: Number
+    Description: Initial storage size for database in GB
+  
+  DBInstanceType:
+    Type: String
+    Description: Instance class for database (eg. "db.m7g.large")
```
Comment (tayritenour) [inline harness/determined/deploy/aws/templates/simple-rds.yaml:109]: lets also change the "class" to "type"

File: harness/determined/deploy/aws/cli.py [hunk 3]
```diff
@@ -437,6 +442,18 @@ def handle_dump_master_config_template(args: argparse.Namespace) -> None:
                     default=constants.defaults.DB_PASSWORD,
                     help="password for master database",
                 ),
+                Arg(
+                    "--db-instance-type",
+                    type=str,
+                    default=constants.defaults.DB_INSTANCE_TYPE,
+                    help="instance class for master database",
```
Comment (tayritenour) [inline harness/determined/deploy/aws/cli.py:449]: "class" -> "type"

**Review by tayritenour — APPROVED** [review]
LGTM

**Pull request merged**